app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -540,6 +547,71 @@ class HomeActivityTest {
     }
   }
 
+  @Test
+  fun testHomeActivity_allTopicsCompleted_hidesPromotedStories() {
```

Comment:
I think so, tests added

Reply:
Actually it's still unclear what alternatives are to the recently-played stories so I'm not sure what the case is for no story progress -- I only added a test for the recently-played stories for partial progress

Reply:
Per discussion, reminder to add 'All Topics' tests since if we have behaviors that we need to work, we should have tests for them.

Reply:
Tests for no progress / all topics are added. It's not the most robust test since the helper leverages the fact that profile #2 is initialized but never accessed, but I couldn't find a simple way to add a new profile and also login (get its internal ID) without having to add observers or some way to get the internal ID from the controller.